tasks/extract-release-notes-images/extract-release-notes-images.yaml

```diff
@@ -0,0 +1,80 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: extract-release-notes-images
+  labels:
+    app.kubernetes.io/version: "0.1.0"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Tekton task to populate the releaseNotes.content.images key in the data.json file
+  params:
+    - name: dataPath
+      description: Path to the JSON string of the merged data to use
+      type: string
+    - name: snapshotPath
+      description: Path to the JSON string of the mapped Snapshot spec in the data workspace
+      type: string
+    - name: commonTags
+      type: string
+      description: Space separated list of common tags to be used when publishing
+  workspaces:
+    - name: data
+      description: The workspace where the data JSON file resides
+  steps:
+    - name: extract-release-notes-images
+      image: quay.io/redhat-appstudio/release-service-utils:cc80f80ba26c89c4979644e2bc7d8ccbbae7eaa1
+      script: |
+        #!/usr/bin/env sh
+        set -ex
+
+        DATA_FILE="$(workspaces.data.path)/$(params.dataPath)"
+        if [ ! -f "${DATA_FILE}" ] ; then
+            echo "No data JSON was provided."
+            exit 1
+        fi
+
+        SNAPSHOT_FILE="$(workspaces.data.path)/$(params.snapshotPath)"
+        if [ ! -f "${SNAPSHOT_FILE}" ] ; then
+            echo "No valid snapshot file was provided."
+            exit 1
+        fi
+
+        # Common vars
+        # Convert space separated list of common tags into an array
+        tags=$(jq -cn --arg tags "$(params.commonTags)" '$tags|split(" ")')
+
+        for component in $(jq -c '.components[]' "${SNAPSHOT_FILE}")
+        do
+            repo=$(jq -r '.repository' <<< $component)
+            deliveryRepo=$(translate-delivery-repo $repo)
+            image=$(jq -r '.containerImage' <<< $component)
+            if [[ "${image}" != *"@sha256"* || $(echo "${image}" | tr -cd ':' | wc -c) -ne 1 ]] ; then
+                echo "Failed to extract sha256 tag from ${image}. Exiting with failure"
+                exit 1
+            fi
+            sha=$(echo "${image}" | cut -d ':' -f 2)
+            # amd64 for example
+            architecture=$(skopeo inspect --no-tags "docker://${image}" | jq -r '.Architecture')
+            # containerImage should be of the form registry.redhat.io/foo/bar@sha256:abcde
+            # This value will be used as the basis for the example values that follow
+            containerImage="${deliveryRepo}@sha256:${sha}"
+            # repository should be foo/bar
+            repository=$(echo ${containerImage} | cut -d '/' -f 2- | cut -d '@' -f 1)
+            # purl should be pkg:oci/foo@sha256:abcde?repository_url=registry.redhat.io/foo/bar
```

Comment:
@scoheb can you check these fields? I took them from the ticket so I don't think they are wrong, but any typos there would've translated over to here. For example, the ticket has oci in the purl when that isn't in the containerImage at all so I assumed the oci part is hardcoded

Reply:
looks good. the real test will be when we run this with real components.

Reply:
Thanks!